src/main/java/mate/academy/MergeSortAction.java

```diff
@@ -1,8 +1,61 @@
 package mate.academy;
 
-public class MergeSortAction {
+import java.util.Arrays;
+import java.util.concurrent.RecursiveAction;
+
+public class MergeSortAction extends RecursiveAction {
+    private static final int THRESHOLD = 2;
+
+    private final int[] array;
+    private final int start;
+    private final int end;
 
     public MergeSortAction(int[] array) {
+        this(array, 0, array.length);
+    }
+
+    private MergeSortAction(int[] array, int start, int end) {
+        this.array = array;
+        this.start = start;
+        this.end = end;
+    }
+
+    @Override
+    protected void compute() {
+        if (end - start <= THRESHOLD) {
+            Arrays.sort(array, start, end);
+        } else {
+            int middle = (start + end) / 2;
+            MergeSortAction left = new MergeSortAction(array, start, middle);
+            MergeSortAction right = new MergeSortAction(array, middle, end);
```

Comment:
So it will more or less look like this in here:

```suggestion
            MergeSortAction left = new MergeSortAction(someArray);
            MergeSortAction right = new MergeSortAction(anotherArray);
```